modus/overview.mdx

```diff
@@ -8,50 +8,72 @@ sidebarTitle: "Overview"
+| WebAssembly powered runtime | A small, portable, and highly efficient execution engine for deploying agents anywhere           |
+| Flexible tool support       | Consume Model Context Protocol (MCP) servers or build your own custom tools to power your agents |
+| Triggered agent actions     | Define and trigger specific actions for your agents to perform                                   |
+| Built-in agent memory       | Support for both short-term and long-term agent memory, powered by knowledge graphs              |
+| Agent operations            | Built-in inference logging and traces for agent-level observability                              |
+| Serverless scale-to-zero    | Each agent spins up on demand and scales to zero in idle state                                   |
+| Fully open source           | No vendor lock-in, but designed for low cold-start times and rapid scaling                       |
 
 <Note>
   Modus is a multi-language framework. It currently includes support for Go and
   AssemblyScript, a WebAssembly compatible TypeScript-like language. Additional
   language support is in development.
 </Note>
 
-## What is Modus good for? {/* vale Google.Contractions = NO */}
+## When to use Modus
 
-We designed Modus primarily as a general-purpose app framework, it just happens
-to treat models as a first-class component. With Modus you can use models, as
-appropriate, without additional complexity.
+### Rapid prototyping to scaling with confidence
 
-However, Modus is best for apps that require sub-second response times. We've
-made trade-offs to optimize for speed and simplicity.
+Start with one-line agent definitions using **AgentKit** - no infrastructure
+setup, no complex wiring. You get streaming UI, context management, and logging
+out of the box. Once your agent logic is solid, deploy via the same framework.
+Because agents are WebAssembly based and serverless, they incur zero cost when
+idle and start fast under load.
 
-For more inspiration, check out the
-[Modus recipes](https://github.com/hypermodeinc/modus-recipes).
+### Language-agnostic logic, safely
+
+With WebAssembly's secure isolation environment, define agent behavior in Go or
+AssemblyScript. Compile to WebAssembly and Modus runs it securely. Each agent
+runs as a lightweight actor: self-contained, message-driven, and supervised by
+the runtime.
+
+### Persistent context beyond sessions
+
+For use-cases like multi-day workflows, personal assistants, tutoring, or
+domain-specific agents (for example ticketing systems), long-term context is
+crucial. Modus provides it without requiring a dedicated database layer. Modus'
+built-in short-term memory handles conversational state and context within
+agents. Connect to long-term graph memory for externalized and persisted, but
+serverless-friendly memory. You gain full "recall" - across sessions and
+workflows- without maintaining stateful servers.
```

Comment:
```suggestion
For use-cases like multi-day workflows, personal assistants, tutoring, or
domain-specific agents (for example ticketing systems), long-term context is
crucial. Modus provides it without requiring a dedicated database layer. Modus'
built-in short-term memory handles conversational state and context within
agents. You gain full "recall" - across sessions and
workflows- without maintaining stateful servers.

<Note>
    Connect to long-term graph memory, such as Dgraph, for externalised and persisted, serverless-friendly storage.
</Note>
```

Here the note about connecting to long term storage felt a bit odd, wdyt about putting it in a note?